compiler/rustc_session/src/config.rs

```diff
@@ -956,7 +956,7 @@ fn default_configuration(sess: &Session) -> CrateConfig {
     ret.reserve(7); // the minimum number of insertions
     // Target bindings.
     ret.insert((sym::target_os, Some(Symbol::intern(os))));
-    for fam in &sess.target.families {
+    for fam in sess.target.families.iter() {
```

Comment:
Unnecessary change?

Reply:
Unfortunately, no because:
```text
error[E0277]: `&Cow<'_, [Cow<'static, str>]>` is not an iterator
   --> compiler/rustc_session/src/config.rs:959:16
    |
959 |     for fam in &sess.target.families {
    |                ^^^^^^^^^^^^^^^^^^^^^ `&Cow<'_, [Cow<'static, str>]>` is not an iterator
    |
    = help: the trait `Iterator` is not implemented for `&Cow<'_, [Cow<'static, str>]>`
    = note: required because of the requirements on the impl of `IntoIterator` for `&Cow<'_, [Cow<'static, str>]>`
```
But still changed to `as_ref()` instead of `iter()`.